imports/wasi_snapshot_preview1/wasi_stdlib_test.go

```diff
@@ -396,3 +404,59 @@ func testSock(t *testing.T, bin []byte) {
 	require.NoError(t, err)
 	require.Equal(t, "wazero\n", console)
 }
+
+func Test_HTTP(t *testing.T) {
+	toolchains := map[string][]byte{}
+	if wasmGotip != nil {
+		toolchains["gotip"] = wasmGotip
+	}
+
+	for toolchain, bin := range toolchains {
+		toolchain := toolchain
+		bin := bin
+		t.Run(toolchain, func(t *testing.T) {
+			testHTTP(t, bin)
+		})
+	}
+}
+
+func testHTTP(t *testing.T, bin []byte) {
+	sockCfg := experimentalsock.NewConfig().WithTCPListener("127.0.0.1", 0)
+	ctx := experimentalsock.WithConfig(testCtx, sockCfg)
+	// Set context to one that has an experimental listener that logs all host functions.
+	ctx = context.WithValue(ctx, experimental.FunctionListenerFactoryKey{},
+		logging.NewHostLoggingListenerFactory(os.Stdout, logging.LogScopeAll))
+
+	moduleConfig := wazero.NewModuleConfig().
+		WithSysWalltime().WithSysNanotime(). // HTTP middleware uses both clocks
+		WithArgs("wasi", "http")
+	tcpAddrCh := make(chan *net.TCPAddr, 1)
+	ch := make(chan string, 1)
+	go func() {
+		ch <- compileAndRunWithPreStart(t, ctx, moduleConfig, bin, func(t *testing.T, mod api.Module) {
+			tcpAddrCh <- requireTCPListenerAddr(t, mod)
+		})
+	}()
+	tcpAddr := <-tcpAddrCh
+
+	// Give a little time for _start to complete
+	sleepALittle()
+
+	// Now, send a POST to the address which we had pre-opened.
+	body := bytes.NewReader([]byte("wazero"))
+	req, err := http.NewRequest(http.MethodPost, "http://"+tcpAddr.String(), body)
+	require.NoError(t, err)
+
+	// TODO: test hangs here
```

Comment:
here's the hang. from the log, the request is never read.

Reply:
it looks like it's stuck in a loop here:

```go
// poll/fd_unix.go
func (fd *FD) Accept() (int, syscall.Sockaddr, string, error) {
	...
	for {
		...
		case syscall.EAGAIN:
			if fd.pd.pollable() {
				if err = fd.pd.waitRead(fd.isFile); err == nil {
					continue
				}
			}
		...
```

Reply:
ok I am now calling `SetDeadline()`, catching the timeout error, and returning EAGAIN instead. This _I think_ causes the Wasm module to do the loop above 😬 so the wasm module is effectively unstuck but I think it's still looping until the connection becomes available; OTOH this lets other goroutines interleave; in fact I can see a poll_oneoff happening:

```
	==> wasi_snapshot_preview1.clock_time_get(id=monotonic,precision=0)
	<== (timestamp=81477104823666,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
	<== (fd=,errno=EAGAIN)
	==> wasi_snapshot_preview1.poll_oneoff(in=21422080,out=21446656,nsubscriptions=2)
	<== (nevents=2,errno=ESUCCESS)
	==> wasi_snapshot_preview1.sock_accept(fd=3,flags=)
```